src/engine/controls/cuecontrol.cpp

```diff
@@ -519,16 +534,13 @@ void CueControl::reloadCuesFromTrack() {
     double intro = m_pIntroStartPosition->get();
 
     // Make track follow the updated cues.
-    SeekOnLoadMode seekOnLoadMode = getSeekOnLoadMode();
-    if (seekOnLoadMode == SEEK_ON_LOAD_DEFAULT) {
-        if ((trackAt == TrackAt::Cue || wasTrackAtZeroPos) && cue != -1.0 && isCueRecallEnabled()) {
-            seekExact(cue);
-        }
-    } else if (seekOnLoadMode == SEEK_ON_LOAD_MAIN_CUE) {
+    SeekOnLoadMode seekOnLoadMode = getSeekOnLoadPreference();
+
+    if (seekOnLoadMode == SeekOnLoadMode::MainCue) {
         if ((trackAt == TrackAt::Cue || wasTrackAtZeroPos) && cue != -1.0) {
```

Comment:
Can we put this off? I am not trying to refactor all the code. This PR has already had to deal with lots of merge conflicts because of making big changes and sitting unmerged for months.

Reply:
It was introduced in this PR that we use kNoTrigger, originally introduced for looping  also for cue points. It is ok for me to reuse this constant for cue here, but that means to update all occurences. Without this, there is a risk of errors in case we decide later to move a way from the -1 that is unfortunate a valid cue point on the lead in.
It would be also ok to postpone this change into a separate PR. In general this is the best strategy anyway, cut out trivial refactorings and merge them before a feature PR.